src/azure-cli/azure/cli/command_modules/keyvault/_params.py

```diff
@@ -211,29 +211,51 @@ class CLIJsonWebKeyOperation(str, Enum):
     # endregion
 
     # region KeyVault Secret
-    with self.argument_context('keyvault secret set') as c:
-        c.argument('content_type', options_list=['--description'], help='Description of the secret contents (e.g. password, connection string, etc)')
-        c.attributes_argument('secret', SecretAttributes, create=True)
-
-    with self.argument_context('keyvault secret set', arg_group='Content Source') as c:
-        c.argument('value', options_list=['--value'], help="Plain text secret value. Cannot be used with '--file' or '--encoding'", required=False)
-        c.extra('file_path', options_list=['--file', '-f'], type=file_type, help="Source file for secret. Use in conjunction with '--encoding'", completer=FilesCompleter())
-        c.extra('encoding', arg_type=get_enum_type(secret_encoding_values, default='utf-8'), options_list=['--encoding', '-e'], help='Source file encoding. The value is saved as a tag (`file-encoding=<val>`) and used during download to automatically encode the resulting file.')
-
-    with self.argument_context('keyvault secret set-attributes') as c:
-        c.attributes_argument('secret', SecretAttributes)
-
-    with self.argument_context('keyvault secret download') as c:
-        c.argument('file_path', options_list=['--file', '-f'], type=file_type, completer=FilesCompleter(), help='File to receive the secret contents.')
-        c.argument('encoding', arg_type=get_enum_type(secret_encoding_values), options_list=['--encoding', '-e'], help="Encoding of the secret. By default, will look for the 'file-encoding' tag on the secret. Otherwise will assume 'utf-8'.", default=None)
-
-    for scope in ['backup', 'restore']:
-        with self.argument_context('keyvault secret {}'.format(scope)) as c:
-            c.argument('file_path', options_list=['--file', '-f'], type=file_type, completer=FilesCompleter(), help='File to receive the secret contents.')
-
-    for scope in ['list', 'list-deleted', 'list-versions']:
-        with self.argument_context('keyvault secret {}'.format(scope)) as c:
-            c.argument('maxresults', options_list=['--maxresults'], type=int)
+    for item in ['set']:
```

Comment:
Why is `set` extracted to an one-item array?

Reply:
I

> Why is `set` extracted to an one-item array?

I intend to make the code pieces more consistent like the following template:
```
with item in {command_list}:
    with self.argument_context('keyvault secret {}'.format(item)) as c:
         ...
```

But if `len(command_list) == 1`, I'm not sure what is the best practice, do we have any guiding on this?

Reply:
We don't have guideline on this. Using a loop is simply for eliminating duplication. For single item array, I prefer to unwrap it to improve the readability. But it's up to you. 😉

Reply:
Thanks, I'd like to keep this until we have further strict guideline on this.